modules/apps/portal-configuration/portal-configuration-upgrade-impl/src/main/java/com/liferay/portal/configuration/upgrade/internal/PrefsPropsToConfigurationUpgradeHelperImpl.java

```diff
@@ -54,14 +55,34 @@ public void mapConfigurations(
 			Class<?> configurationClass, KeyValuePair... keyValuePairs)
 		throws Exception {
 
-		String filterString = StringBundler.concat(
-			"(", Constants.SERVICE_PID, "=", configurationClass.getName(), ")");
+		mapConfigurations(0, configurationClass, keyValuePairs);
+	}
 
-		Configuration[] configurations = _configurationAdmin.listConfigurations(
-			filterString);
+	@Override
+	public void mapConfigurations(
+			long companyId, Class<?> configurationClass,
+			KeyValuePair... keyValuePairs)
+		throws Exception {
 
-		if (configurations != null) {
-			return;
+		if (companyId > 0) {
+			Object object = _configurationProvider.getCompanyConfiguration(
+				configurationClass, companyId);
+
+			if (object != null) {
+				return;
+			}
```

Comment:
Hey @ChrisKian, unfortunately I don't think that this block will work. To my knowledge, the `object` here will never be null. In this case the goal is to check to see if it exists, so we probably want to use the filter string approach instead of the ConfigurationProvider method. We can still use ConfigurationProvider to save the upgraded configuration below.

You can copy the logic from ConfigurationProvider here at [`saveFactoryConfiguration`](https://github.com/liferay/liferay-portal/blob/31459ed4ca6f158bf206b493da30c6c303070d38/modules/apps/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationProviderImpl.java#L331) which in turn calls [`_getFactoryConfiguration`](https://github.com/liferay/liferay-portal/blob/31459ed4ca6f158bf206b493da30c6c303070d38/modules/apps/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationProviderImpl.java#L273).